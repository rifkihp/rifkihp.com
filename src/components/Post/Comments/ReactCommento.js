// @flow
import * as React from 'react';

export default class ReactCommento extends React.PureComponent<{||}> {
  _script: ?HTMLScriptElement = null;

  componentDidMount() {
    const script = document.createElement('script');
    script.async = true;
    script.type = 'text/javascript';
    script.src = '//cdn.commento.io/js/commento.js';
    script.setAttribute('data-css-override', '/commento.min.css');

    const head = document.getElementsByTagName('head')[0];
    head.appendChild(script);
    this._script = script;
  }

  componentWillUnmount() {
    if (this._script) {
      this._script.remove();
      this._script = null;
    }
  }

  render() {
    return <div id="commento" />;
  }
}
