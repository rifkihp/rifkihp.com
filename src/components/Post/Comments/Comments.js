// @flow
import * as React from 'react';

import ReactCommento from './ReactCommento';

type State = {|
  +show: boolean,
|};

export default class Comments extends React.PureComponent<{||}, State> {
  state = { show: false };

  timeout: ?TimeoutID;

  componentDidMount() {
    window.addEventListener('scroll', this.onScroll);
    this.timeout = setTimeout(this.onScroll, 5000);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.onScroll);
    clearTimeout(this.timeout);
  }

  onScroll = () => {
    this.setState({ show: true });
    window.removeEventListener('scroll', this.onScroll);
  };

  render() {
    if (!this.state.show) {
      return <div style={{ paddingBottom: '320px' }} />;
    }

    return <ReactCommento />;
  }
}
