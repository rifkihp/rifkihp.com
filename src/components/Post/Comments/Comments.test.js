import React from 'react';
import renderer from 'react-test-renderer';

import Comments from './Comments';

describe('Comments', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Comments />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
