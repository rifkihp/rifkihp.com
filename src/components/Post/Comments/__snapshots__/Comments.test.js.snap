// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders correctly 1`] = `
<div
  style={
    Object {
      "paddingBottom": "320px",
    }
  }
/>
`;
